### PR TITLE
Return 200 to Office making WebDAV requests

### PR DESCRIPTION
**Pull request by thewilkybarkid**
As mentioned in ELPP-3522.

Not tested anywhere yet.

### CODE REVIEW

File: src/buildercore/fastly/vcl/office-webdav-200.vcl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+if(
+    req.http.User-Agent ~ "Microsoft Office Protocol Discovery"
+    ||
+    req.http.User-Agent ~ "Microsoft Office Existence Discovery"
+    ||
+    req.http.User-Agent ~ "Microsoft-WebDAV-MiniRedir"
+    ||
+    (req.http.User-Agent ~ "ms-office" && req.http.User-Agent !~ "Microsoft Outlook")
+) {
+    error 200;
```
Comment (thewilkybarkid) [inline src/buildercore/fastly/vcl/office-webdav-200.vcl]: This should result in a 200 Varnish error page. Alternative is to use a `9xx` code and pick it up in `vcl_error`, but requires another VCL file.

File: src/buildercore/fastly/vcl/office-webdav-200.vcl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+if(
+    req.http.User-Agent ~ "Microsoft Office Protocol Discovery"
```
Comment (giorgiosironi) [inline src/buildercore/fastly/vcl/office-webdav-200.vcl]: this is the list of Microsoft user agents getting 400 recently:
```
$ grep -rh '"400"' 2018-05-* 2018-06-01* | jq .request_user_agent -r | sort | uniq -c | grep Microsoft
      3 Microsoft Office Excel 2013
      1 Microsoft Office Excel 2014
      2 Microsoft Office PowerPoint/15.25.630 (Mac OS/10.13.3; Desktop; zh-CN; NonAppStore; Apple/MacBookAir6,2)
     11 Microsoft Office Protocol Discovery
```

File: src/buildercore/fastly/vcl/office-webdav-200.vcl (hunk 1 above)
Comment (giorgiosironi) [reply src/buildercore/fastly/vcl/office-webdav-200.vcl]: If 200 works (shouldn't be visible to an end user)

**Comment by giorgiosironi** [comment]
Tested:
```
$ curl https://continuumtest-cdn.elifesciences.org/something -H "User-Agent: Microsoft Office Protocol Discovery" -X OPTIONS

<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>200 OK</title>
  </head>
  <body>
    <h1>Error 200 OK</h1>
    <p>OK</p>
    <h3>Guru Mediation:</h3>
    <p>Details: cache-lcy19228-LCY 1528125654 2302949171</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```

**Pull request merged**